WordPress/Classes/ViewRelated/Tools/SettingsMultiTextViewController.h

```diff
@@ -1,13 +1,36 @@
 #import <UIKit/UIKit.h>
 
+
+typedef void (^SettingsMultiTextAction)(void);
+typedef void (^SettingsMultiTextChanged)(NSString * _Nonnull);
+
+
+/// Reusable component that renders a UITextView + Hint onscreen. Useful for Text / Password.
+///
 @interface SettingsMultiTextViewController : UITableViewController
 
-@property (nonatomic, copy) void(^onValueChanged)(NSString *);
-@property (nonatomic, copy) void(^onCancel)();
+/// String to be displayed at the bottom.
+///
+@property (nullable, nonatomic, strong) NSString *hint;
+
+/// Placeholder string to be displayed, in case the text is empty.
+///
+@property (nullable, nonatomic, strong) NSString *placeholder;
+
+/// The raw string (current value) to edit.
+///
+@property (nullable, nonatomic, strong) NSString *text;
+
+/// Boolean indicating whether secure text entry is required, or not.
+///
+@property (nonatomic, assign) BOOL isPassword;
+
+/// Block to be executed on dismiss, if the value was effectively updated.
+///
+@property (nullable, nonatomic, copy) SettingsMultiTextChanged onValueChanged;
 
-- (instancetype)initWithText:(NSString *)text
```

Comment:
Couldn't we keep this init for code base use? BTW this is one of the reasons I don't like to use storyboards :)